### DISCUSSION TITLE
Guide for running the mocha webdriver tests

### DISCUSSION THREAD

**Discussion by tgowing** [appium/appium#19479] [Q&A]
First of all thank you for this amazing project!

We already have the tests written in the following format and we run them with selenium-webdriver:

```js
const { expect } = require("chai");
const { Builder, By, until } = require("selenium-webdriver");
const { globalElementsAndFunctions } = require('../../page-models/global-functions.js');
const csv = require('csv-parser');
const fs = require('fs');

describe("Other Tests", function() {
    this.timeout(500000);
    let driver;
...
```

We run these using `mocha test.js`.

Is there a guide how to run the tests with appium, preferably without changing anything in the tests?

BrowserStack SDK has a CLI utility that integrates nicely with `mocha`: 

```sh
browserstack-node-sdk mocha test.js
```

Does `appium` have something like this?

Thank you very much.

**Comment by jlipps** [comment]
You can run Appium tests in exactly the same way. If you use the selenium client you just might not have access to certain Appium functionality, though most of that is now accessible via executeScript anyway.
